query-translator/src/types.rs

```diff
@@ -25,37 +27,57 @@ use mentat_sql::{
     QueryBuilder,
     QueryFragment,
     SQLiteQueryBuilder,
+    SQLQuery,
 };
 
 //---------------------------------------------------------
 // A Mentat-focused representation of a SQL query.
 
-enum ColumnOrExpression {
+pub enum ColumnOrExpression {
     Column(QualifiedAlias),
-    Integer(i64),             // Because it's so common.
+    Integer(i64),            // Because it's so common.
+    Value(TypedValue),       // TODO: replace with a SQLite value.
```

Comment:
I vacillated on this. As you saw earlier, I simply allowed the SQL layer to accept a `TypedValue`, which avoids _this_ layer eventually having to worry about `rusqlite::ToSql`. I should probably remove this comment. The other approach is to have this be `ToSql` (or similar), and translate `TypedValue` at the point of constructing a `SelectQuery`. Do you have a preference, @ncalexan?